docs/usageScenarios.md

```diff
@@ -0,0 +1,66 @@
+
+From that point on they all have full traceability of all the changes made, and they are able to roll it back easily by reverting commits. They can review the changes simply by looking at the git diffs on commits.
+
+**Getting there and adopting Jenkins Configuration as Code** is also quite easy for the devops team. They will need to install the plugin, and go the the Configuration as Code menu in Manage Jenkins, and use the export function to create the initial YAML files that represent their current global configuration and plugin's installed.
+
+When the files are in git, they can point to the file in the repository in the Configuration as Code menu, and click reload so Jenkins updates it's configuration accordingly.
+
+
+_Migration step-by-step guide:_ For the exact steps, see this part of the migration guide - TBD when all relevant issues fixed:
+
+  * [#6 about reloading configuration when changed](https://github.com/jenkinsci/configuration-as-code-plugin/issues/6)
+  * [#7 about plugin installation support](https://github.com/jenkinsci/configuration-as-code-plugin/issues/7)
+  * [#10 related to hierarchy and re-use of configuration](https://github.com/jenkinsci/configuration-as-code-plugin/issues/10)
+  * [#32 where to draft a migration to Jenkins Configuration as Code guide](https://github.com/jenkinsci/configuration-as-code-plugin/issues/32)
+  * [#65 helping users migrate by export existing configuration](https://github.com/jenkinsci/configuration-as-code-plugin/issues/65)
+
+
+
+## Jenkins master already in docker
+
+A small full stack developer team have their single Jenkins master running in a container and they orchestrate and deploy it using one of the many container orchestration tools.
+Their Jenkins global configuration is the only piece not under version control, so they depend on the current data in `JENKINS_HOME` to persist the Jenkins global configuration and be able to redeploy the Jenkins master with existing installed plugins. They do not care about their historic build data.
+
+**With Jenkins configuration as code plugin** they can get their missing configuration under version control as well and their installed plugin combinations and for this little full stack developer team it means they can always just redeploy their Jenkins should something be wrong or failing, and their container orchestration tool will keep their Jenkins master always running for them completely automatically.
+
+**Migrating to use the Jenkins Configuration as Code plugin** is easy if they can live with doing just few manual steps before it is all automated. They need to install the Jenkins Configuration as Code plugin on their current master, then use the export functionality to create the `plugins.yml` and `jenkins.yml` file for them, which they need to add to a git repository.
```

Comment:
I'm wondering if maybe we could also add here a case where we use plugins.txt to install the plugins - including our plugin and feed docker with jenkins.yaml location in an environment variable plugin recognises? is the same case you're describing or separate one?

Reply:
Yes the same case, but couldn't mention the docker example command as this team uses an orchestration tool so they describe this in a deployment file...  but conceptually it is the same setup.

Reply:
Updated the description to elaborate the difference between plugins.txt and plugins.yml and why we need a image to support their initial setup.

As part of this issue by the way I wrote some issues on things we need: #65 and now #70